src/containers/pages/IRS/assignments/index.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useState } from 'react';
+import { Helmet } from 'react-helmet';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { Col, Row } from 'reactstrap';
+import { Store } from 'redux';
+import HeaderBreadcrumb from '../../../../components/page/HeaderBreadcrumb/HeaderBreadcrumb';
+import Loading from '../../../../components/page/Loading';
+import { useContextCodes } from '../../../../configs/settings';
+import {
+  ASSIGN_IRS_PLAN_URL,
+  DRAFT_IRS_PLAN_URL,
+  HOME,
+  HOME_URL,
+  IRS_PLAN_TYPE,
+  IRS_PLANS,
+  REPORT_IRS_PLAN_URL,
+} from '../../../../constants';
+import { OpenSRPService } from '../../../../services/opensrp';
+import IRSPlansReducer, {
+  reducerName as IRSPlansReducerName,
+} from '../../../../store/ducks/generic/plans';
+import {
+  extractPlanRecordResponseFromPlanPayload,
+  fetchPlanRecords,
+  getPlanRecordsArray,
+  InterventionType,
+  PlanPayload,
+  PlanRecord,
+  PlanRecordResponse,
+  PlanStatus,
+} from '../../../../store/ducks/plans';
+
+/** register the plan definitions reducer */
+reducerRegistry.register(IRSPlansReducerName, IRSPlansReducer);
+
+const OpenSrpPlanService = new OpenSRPService('plans');
+
+/** interface for PlanAssignmentsListProps props */
+interface PlanAssignmentsListProps {
+  fetchPlans: typeof fetchPlanRecords;
+  plans: PlanRecord[];
+}
+
+/** Simple component that loads plans and allows you to manage plan-jurisdiciton-organization assignments */
+const IRSAssignmentPlansList = (props: PlanAssignmentsListProps) => {
+  const { fetchPlans, plans } = props;
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const pageTitle: string = IRS_PLANS;
+
+  const breadcrumbProps = {
+    currentPage: {
+      label: pageTitle,
+      url: REPORT_IRS_PLAN_URL,
+    },
+    pages: [
+      {
+        label: HOME,
+        url: HOME_URL,
+      },
+    ],
+  };
+
+  /** async function to load the data */
+  async function loadData() {
+    try {
+      setLoading(plans.length < 1); // only set loading when there are no plans
+      OpenSrpPlanService.list()
+        .then(planResults => {
+          // filter for IRS plans
+          const irsPlans = planResults.filter((p: PlanPayload) => {
+            for (const u of p.useContext) {
+              if (u.code === useContextCodes[0]) {
+                if (u.valueCodableConcept === IRS_PLAN_TYPE) {
+                  return true;
+                } else {
+                  return false;
+                }
+              }
+            }
+            return false;
+          });
+          const irsPlanRecords: PlanRecordResponse[] = irsPlans.map(
+            extractPlanRecordResponseFromPlanPayload
+          );
+          return fetchPlans(irsPlanRecords);
+        })
+        .catch(err => {
+          // console.log('ERR', err)
+        });
+    } catch (e) {
+      // do something with the error?
+    } finally {
+      setLoading(false);
```

Comment:
might result in a `setState on unmounted component error`, possible solution would be to have `loading` as a computed variable of the plans length. i.e. `loading = plans.length < 1`

Reply:
@p-netm thanks, this was really insightful! I ended up moving this line up just after the OpenSRP call (to which I added `await`) so it should no longer be in a block that will be called asynchronously. To your point, we should NEVER be setting component state in an asynchronous block or callback. 

👍* 💯